go/v1beta1/storage/esutil/client_test.go

```diff
@@ -0,0 +1,981 @@
+				expectedUpdateRequest.Refresh = "false"
+			})
+
+			It("should not refresh the index after updating the document", func() {
+				Expect(transport.ReceivedHttpRequests[0].URL.Query().Get("refresh")).To(Equal("false"))
+			})
+		})
+	})
+})
+
+func createRandomOccurrence() *pb.Occurrence {
+	return &pb.Occurrence{
+		Name: fake.LetterN(10),
+		Resource: &pb.Resource{
+			Uri: fake.LetterN(10),
+		},
+		NoteName:    fake.LetterN(10),
+		Kind:        common_go_proto.NoteKind_NOTE_KIND_UNSPECIFIED,
+		Remediation: fake.LetterN(10),
+		Details:     nil,
+		CreateTime:  ptypes.TimestampNow(),
+	}
+}
+
+func createRandomOccurrences(l int) []*pb.Occurrence {
+	var result []*pb.Occurrence
+	for i := 0; i < l; i++ {
+		result = append(result, createRandomOccurrence())
+	}
+
+	return result
+}
+
+func createRandomSearch() *EsSearch {
+	return &EsSearch{
+		Query: &filtering.Query{
+			Term: &filtering.Term{
+				fake.LetterN(10): fake.LetterN(10),
+			},
+		},
+		Sort: map[string]EsSortOrder{
+			fake.LetterN(10): EsSortOrderDescending,
+		},
+		Collapse: &EsSearchCollapse{
+			Field: fake.LetterN(10),
+		},
+	}
+}
+
+func createRandomSearches(l int) []*EsSearch {
+	var result []*EsSearch
+	for i := 0; i < l; i++ {
+		result = append(result, createRandomSearch())
+	}
+
+	return result
+}
+
+func createRandomDocumentIds(l int) []string {
+	var result []string
+	for i := 0; i < l; i++ {
+		result = append(result, fake.LetterN(10))
+	}
+
+	return result
+}
+
+func structToJsonBody(i interface{}) io.ReadCloser {
+	b, err := json.Marshal(i)
+	Expect(err).ToNot(HaveOccurred())
+
+	return ioutil.NopCloser(strings.NewReader(string(b)))
+}
+
+// helper functions for _bulk requests
+
+func createEsBulkOccurrenceIndexResponse(occurrences []*pb.Occurrence, errs []error) *EsBulkResponse {
+	var (
+		responseItems     []*EsBulkResponseItem
+		responseHasErrors = false
+	)
+	for i := range occurrences {
+		var (
+			responseErr  *EsIndexDocError
+			responseCode = http.StatusCreated
+		)
+		if errs[i] != nil {
+			responseErr = &EsIndexDocError{
+				Type:   fake.LetterN(10),
+				Reason: fake.LetterN(10),
+			}
+			responseCode = http.StatusInternalServerError
+			responseHasErrors = true
+		}
+
+		responseItems = append(responseItems, &EsBulkResponseItem{
+			Index: &EsIndexDocResponse{
+				Id:     fake.LetterN(10),
+				Status: responseCode,
+				Error:  responseErr,
+			},
+		})
+	}
+
+	return &EsBulkResponse{
+		Items:  responseItems,
+		Errors: responseHasErrors,
+	}
+}
+
+// parseEsNDJSONRequestBodyWithProtobufs parses a request body in ndjson format
+// each line of the body is assumed to be properly formatted JSON
+// every odd line is assumed to be a regular JSON structure that can be unmarshalled via json.Unmarshal
+// every even line is assumed to be a JSON structure representing a protobuf message, and will be unmarshalled using protojson.Unmarshal
+func parseEsNDJSONRequestBodyWithProtobufs(body io.ReadCloser, structs []interface{}) {
```

Comment:
I think ndjson isn't unique to Elasticsearch, and removing it also makes the function name a little easier to grok. I thought this was a bad refactor/rename at first. 
```suggestion
func parseNDJSONRequestBodyWithProtobufs(body io.ReadCloser, structs []interface{}) {
```